packages/jest-cli/src/init/generate_config_file.ts

```diff
@@ -81,10 +87,15 @@ const generateConfigFile = (
     }
   }
 
-  return (
+  const configDocMessage =
     '// For a detailed explanation regarding each configuration property, visit:\n' +
-    '// https://jestjs.io/docs/en/configuration.html\n\n' +
-    (generateEsm ? 'export default {\n' : 'module.exports = {\n') +
+    '// https://jestjs.io/docs/en/configuration.html\n\n';
+
+  return (
+    configDocMessage +
+    (useTypescript || generateEsm
+      ? 'export default {\n'
```

Comment:
should we include the `types` import? not sure

Reply:
Good question, I thought so but then I got to the point in which I needed the `@types/jest` to use the `Config.InitialOptions` type.

I will be nice to have it but I didn't know the best practice. Do you have any idea?

Reply:
We could possibly insert a code comment at the top for it? `@jest/types` tho, not `@types/jest` 🙂

Reply:
Oops, yes `@jest/types` 😄 

Do you think it will be a good idea to add a comment with the `const config: Config.InitialOptions = ...`? Or write a comment with indications?

Reply:
Not sure...

```ts
/**
If you want to typecheck the config, you do this:

import type { Config } from '@jest/types';

const config: Config.InitialOptions = { ... };

export default config;
*/
```

?

Reply:
@SimenB Agree, maybe we can indicate to the documentation? Something like this:

```
/**
 * For detailed explanation regarding the type check, visit:
 * https://jestjs.io/docs/en/configuration.html
 */
```

Reply:
Yep 👍

Reply:
@SimenB There was a similar message for the `.js` file so I had to adapt it.

For `.js`:
```js
/*
 * For a detailed explanation regarding each configuration property, visit:
 * https://jestjs.io/docs/en/configuration.html
 */
```

and for `.ts`:
```js
/*
 * For a detailed explanation regarding each configuration property and type check, visit:
 * https://jestjs.io/docs/en/configuration.html
 */
```